stingray/lightcurve.py

```diff
@@ -175,3 +209,163 @@ def rebin_lightcurve(self, dt_new, method='sum'):
+            it can be a list like ``[xmin, xmax, ymin, ymax]`` or any other
+            acceptable argument for `matplotlib.pyplot.axis()` function.
+
+        title : str, default None
+            The title of the plot.
+
+        save : boolean, optional (default=False)
+            If True, save the figure with specified filename.
+
+        filename : str
+            File name of the image to save. Depends on the boolean ``save``.
+        """
+        try:
+            import matplotlib.pyplot as plt
+        except ImportError:
+            raise ImportError("Matplotlib required for draw()")
+        except RuntimeError:
+            print("Matplotlib unable to open display.")
+            raise
+
+        fig = plt.figure()
+        fig = plt.plot(self.time, self.counts, 'rx')
+
+        if labels is not None:
+            try:
+                plt.xlabel(labels[0])
+                plt.ylabel(labels[1])
+            except TypeError:
+                utils.simon("``labels`` must be either a list or tuple with "
+                            "x and y labels.")
+                raise
+            except IndexError:
+                utils.simon("``labels`` must have two labels for x and y "
+                            "axes.")
+                # Not raising here because in case of len(labels)==1, only
+                # x-axis will be labelled.
+
+        if axis is not None:
+            plt.axis(axis)
+
+        plt.title(title)
+
+        if save:
+            if filename is None:
+                self.savefig('out.png')
+            else:
+                self.savefig(filename)
+
+    def savefig(self, filename, **kwargs):
+        """
+        Save the Lightcurve figure plotted by Matplotlib.
+
+        Note : This function is supposed to be used after the ``draw``
+        function. Otherwise it will save a blank image with no plot.
+
+        Parameters
+        ----------
+        filename : str
+            The name of the image file. Extension must be specified in the
+            file name. For example filename with `.png` extension will give a
+            rasterized image while `.pdf` extension will give a vectorized
+            output.
+
+        kwargs : keyword arguments
+            Keyword arguments to be passed to ``savefig`` function of
+            ``matplotlib.pyplot``. For example use `bbox_inches='tight'` to
+            remove the undesirable whitepace around the image.
+        """
+
+        try:
+            import matplotlib.pyplot as plt
+        except ImportError:
+            raise ImportError("Matplotlib required for draw()")
+        except RuntimeError:
+            print("Matplotlib unable to open display.")
+            raise
+
+        if not plt.fignum_exists(1):
+            utils.simon("use ``draw`` function to draw the image first and "
+                        "then use ``savefig`` to save the figure.")
+
+        plt.savefig(filename, **kwargs)
```

Comment:
I would probably pull this method out of the class and stick it into `io.py` or `utils.py` or something, since it's general enough that we might want to use it elsewhere?